src/Microsoft.Health.Dicom.Core/Features/CustomTag/GetCustomTagHandler.cs

```diff
@@ -0,0 +1,32 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using MediatR;
+using Microsoft.Health.Dicom.Core.Features.Validation;
+using Microsoft.Health.Dicom.Core.Messages.CustomTag;
+
+namespace Microsoft.Health.Dicom.Core.Features.CustomTag
+{
+    public class GetCustomTagHandler : IRequestHandler<GetCustomTagRequest, GetCustomTagResponse>
+    {
+        private readonly IGetCustomTagsService _getCustomTagsService;
+
+        public GetCustomTagHandler(IGetCustomTagsService getCustomTagsService)
+        {
+            EnsureArg.IsNotNull(getCustomTagsService, nameof(getCustomTagsService));
+            _getCustomTagsService = getCustomTagsService;
+        }
+
+        public async Task<GetCustomTagResponse> Handle(GetCustomTagRequest request, CancellationToken cancellationToken)
+        {
+            TagPathValidator.Validate(request.CustomTagPath);
```

Comment:
I see most our code follow such pattern where input parameters are checked nullable for any public methods. could you also follow this?

Reply:
Sure, I'll update to validate that.

---
In reply to: [558527662](https://github.com/microsoft/dicom-server/pull/501#discussion_r558527662) [](ancestors = 558527662)